app/dashboard/page.tsx

```diff
@@ -1,11 +1,104 @@
 // Profile page, default dashboard page
 export default function DashboardProfile() {
-  return (
-    <div className="">
-      <PageHeading header="Profile" description="This is how others will see you on the site"/>
+  const { toast } = useToast();
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [bio, setBio] = useState("");
+  const [org, setOrg] = useState("");
+  const [loading, setLoading] = useState(true);
+  const [updating, setUpdating] = useState(false);
+  const isValidEmail = /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/g;
+
+  const handleUpdateProfile = async () => {
+    setUpdating(true);
+    const error = await updateProfile(name, email, bio);
+    setUpdating(false);
+    if (error) {
+      displayErrorToast(error);
+    }
+    else {
+      displaySuccessToast()
+    }
+  }
+
+  const displayErrorToast = (error) => {
+    toast({
+      variant: "destructive",
+      title: "Unable to update profile",
+      description: "Check log for error",
+    })
+  }
 
+  const displaySuccessToast = () => {
+    toast({
+      title: "Profile Successfully Updated",
+    })
+  }
+
+  useEffect(() => {
+    getProfileByUserId().then((res) => {
+      console.log(res);
+      setName(res?.full_name || "");
+      setEmail(res?.email || "");
+      setBio(res?.bio || "");
+      setOrg(res?.organizations?.name || "You are not part of an organization. Go to organization overview to create or join an organization.");
+      setLoading(false);
+    }).catch((error) => (displayErrorToast(error)));
+  }, [])
+
+  return (
+    <div className="flex flex-col 2xl:gap-y-12 gap-y-10">
+      <PageHeading header="Profile" description="This is how others will see you on the site" />
+      {loading ?
+        <div className="text-center 2xl:mt-72 mt-52 text-darkmaroon">
+          <div className="text-xl font-medium">Getting Profile data</div>
+          <div className="loadingAnimation"><div></div><div></div><div></div></div>
+        </div>
+        :
+        <div className="flex flex-col 2xl:gap-y-12 2xl:w-[1200px] gap-y-10 w-[645px]">
+          {/* Name */}
+          <div className="h-24 text-wrap">
+            <div className="2xl:text-3xl text-2xl font-medium mb-4">
+              Name
+            </div>
+            <input type="text" minLength={4} value={name} onChange={(e) => { setName(e.target.value) }} className="outline-none w-full bg-inherit 2xl:text-lg text-body border-b border-body" />
+            {name.length == 0 && <p className="text-red-800 mt-1 2xl:text-base text-sm">Name cannot be empty</p>}
+          </div>
+          {/* Email */}
+          <div className="h-24 text-wrap">
+            <div className="2xl:text-3xl text-2xl font-medium mb-4">
+              Display Email
+            </div>
+            <div className="outline-none w-full bg-inherit 2xl:text-lg text-body border-b border-body" >
+              {email}
+            </div>
+          </div>
```

Comment:
Since we decided on not editing email we could move it below bio to keep all the editable things together for better UX. Also could delete line under email to show user its not editable